quipucords/scanner/network/runner/roles/ifconfig/tasks/main.yml

```diff
@@ -5,13 +5,13 @@
     internal_host_started_processing_role: "ifconfig"
 
 - name: gather ifconfig.mac-addresses fact (no sudo)
-  raw: /sbin/ifconfig -a |  grep -o -E '([[:xdigit:]]{1,2}:){5}[[:xdigit:]]{1,2}'
+  raw: echo $(if [ -f /sbin/ifconfig ] ; then /sbin/ifconfig -a; else /sbin/ip address show; fi) | grep -o -E '([[:xdigit:]]{1,2}:){5}[[:xdigit:]]{1,2}' | grep -v '00:00:00:00:00:00' | grep -v 'ff:ff:ff:ff:ff:ff'
   register: internal_ifconfig_mac_addresses_cmd
   ignore_errors: yes
   when: '(not user_has_sudo) and internal_have_ifconfig'
```

Comment:
I think I'm missing something. Shouldn't we need to change `internal_have_ifconfig` to accommodate this?

Reply:
Double ugh. This variable logic is such a complex mess. I hesitate to add to it, but it might be a necessary evil in the short term versus.